lib/core/asynctree.js

```diff
@@ -77,7 +77,6 @@ class AsyncTree extends EventEmitter{
         let abortOnFailure = false;
 
         if (result instanceof Error) {
-          node.reject(result);
```

Comment:
was this added recently, and something that was working fine before?
Because i believe the issue reported is a regression right ?

Reply:
No this was not added recently but as and when the node promise is rejected, the user needs to handle the rejection in the script or the scripts ends in between.

Reply:
Okay, I am confused, the issue raised is a regression, meaning it was working fine until some change that we did. So user hasnt changed their script, they just upgraded the version right ?

Reply:
Yes, perhaps I need to dig up further how the rejection was being handled in previous versions.

Reply:
Well, whatever it is, it's fixed. Great work Ravi :)